--- lindera-dictionary v3.0.7 (public API)
+++ lindera-dictionary v4 (public API)
@@ -857,6 +857,7 @@
 impl<T: core::ops::deref::Deref<Target = [u8]>> core::convert::From<&'static T> for lindera_dictionary::util::Data
 pub fn lindera_dictionary::util::Data::from(&'static T) -> Self
 pub fn lindera_dictionary::util::mmap_file(&std::path::Path) -> lindera_dictionary::LinderaResult<memmap2::Mmap>
+pub fn lindera_dictionary::util::read_aligned_file(&std::path::Path) -> lindera_dictionary::LinderaResult<rkyv::util::alloc::aligned_vec::AlignedVec<16>>
 pub fn lindera_dictionary::util::read_file(&std::path::Path) -> lindera_dictionary::LinderaResult<alloc::vec::Vec<u8>>
 pub fn lindera_dictionary::util::read_file_with_encoding(&std::path::Path, &str) -> lindera_dictionary::LinderaResult<alloc::string::String>
 pub fn lindera_dictionary::util::write_data<W: std::io::Write>(&[u8], &mut W) -> lindera_dictionary::LinderaResult<()>
@@ -868,18 +869,6 @@
 impl rkyv::traits::Portable for lindera_dictionary::viterbi::ArchivedLexType
 impl<__C: rancor::Fallible + ?core::marker::Sized> bytecheck::CheckBytes<__C> for lindera_dictionary::viterbi::ArchivedLexType where <__C as rancor::Fallible>::Error: rancor::Source
 pub unsafe fn lindera_dictionary::viterbi::ArchivedLexType::check_bytes(*const Self, &mut __C) -> core::result::Result<(), <__C as rancor::Fallible>::Error>
-pub enum lindera_dictionary::viterbi::EdgeType
-pub lindera_dictionary::viterbi::EdgeType::INSERTED
-pub lindera_dictionary::viterbi::EdgeType::KNOWN
-pub lindera_dictionary::viterbi::EdgeType::UNKNOWN
-pub lindera_dictionary::viterbi::EdgeType::USER
-impl core::clone::Clone for lindera_dictionary::viterbi::EdgeType
-pub fn lindera_dictionary::viterbi::EdgeType::clone(&self) -> lindera_dictionary::viterbi::EdgeType
-impl core::default::Default for lindera_dictionary::viterbi::EdgeType
-pub fn lindera_dictionary::viterbi::EdgeType::default() -> lindera_dictionary::viterbi::EdgeType
-impl core::fmt::Debug for lindera_dictionary::viterbi::EdgeType
-pub fn lindera_dictionary::viterbi::EdgeType::fmt(&self, &mut core::fmt::Formatter<'_>) -> core::fmt::Result
-impl core::marker::Copy for lindera_dictionary::viterbi::EdgeType
 pub enum lindera_dictionary::viterbi::LexType
 pub lindera_dictionary::viterbi::LexType::System
 pub lindera_dictionary::viterbi::LexType::Unknown
@@ -912,28 +901,14 @@
 pub lindera_dictionary::viterbi::LexTypeResolver::Unknown
 pub lindera_dictionary::viterbi::LexTypeResolver::User
 #[repr(C)] pub struct lindera_dictionary::viterbi::ArchivedWordEntry where lindera_dictionary::viterbi::WordId: rkyv::traits::Archive, i16: rkyv::traits::Archive, u16: rkyv::traits::Archive
-pub lindera_dictionary::viterbi::ArchivedWordEntry::left_id: <u16 as rkyv::traits::Archive>::Archived
-pub lindera_dictionary::viterbi::ArchivedWordEntry::right_id: <u16 as rkyv::traits::Archive>::Archived
-pub lindera_dictionary::viterbi::ArchivedWordEntry::word_cost: <i16 as rkyv::traits::Archive>::Archived
-pub lindera_dictionary::viterbi::ArchivedWordEntry::word_id: <lindera_dictionary::viterbi::WordId as rkyv::traits::Archive>::Archived
 impl rkyv::traits::Portable for lindera_dictionary::viterbi::ArchivedWordEntry where lindera_dictionary::viterbi::WordId: rkyv::traits::Archive, i16: rkyv::traits::Archive, u16: rkyv::traits::Archive, <lindera_dictionary::viterbi::WordId as rkyv::traits::Archive>::Archived: rkyv::traits::Portable, <i16 as rkyv::traits::Archive>::Archived: rkyv::traits::Portable, <u16 as rkyv::traits::Archive>::Archived: rkyv::traits::Portable
 impl<__C: rancor::Fallible + ?core::marker::Sized> bytecheck::CheckBytes<__C> for lindera_dictionary::viterbi::ArchivedWordEntry where lindera_dictionary::viterbi::WordId: rkyv::traits::Archive, i16: rkyv::traits::Archive, u16: rkyv::traits::Archive, <__C as rancor::Fallible>::Error: rancor::Trace, <lindera_dictionary::viterbi::WordId as rkyv::traits::Archive>::Archived: bytecheck::CheckBytes<__C>, <i16 as rkyv::traits::Archive>::Archived: bytecheck::CheckBytes<__C>, <u16 as rkyv::traits::Archive>::Archived: bytecheck::CheckBytes<__C>
 pub unsafe fn lindera_dictionary::viterbi::ArchivedWordEntry::check_bytes(*const Self, &mut __C) -> core::result::Result<(), <__C as rancor::Fallible>::Error>
 #[repr(C)] pub struct lindera_dictionary::viterbi::ArchivedWordId where u32: rkyv::traits::Archive, bool: rkyv::traits::Archive, lindera_dictionary::viterbi::LexType: rkyv::traits::Archive
-pub lindera_dictionary::viterbi::ArchivedWordId::id: <u32 as rkyv::traits::Archive>::Archived
-pub lindera_dictionary::viterbi::ArchivedWordId::is_system: <bool as rkyv::traits::Archive>::Archived
-pub lindera_dictionary::viterbi::ArchivedWordId::lex_type: <lindera_dictionary::viterbi::LexType as rkyv::traits::Archive>::Archived
 impl rkyv::traits::Portable for lindera_dictionary::viterbi::ArchivedWordId where u32: rkyv::traits::Archive, bool: rkyv::traits::Archive, lindera_dictionary::viterbi::LexType: rkyv::traits::Archive, <u32 as rkyv::traits::Archive>::Archived: rkyv::traits::Portable, <bool as rkyv::traits::Archive>::Archived: rkyv::traits::Portable, <lindera_dictionary::viterbi::LexType as rkyv::traits::Archive>::Archived: rkyv::traits::Portable
 impl<__C: rancor::Fallible + ?core::marker::Sized> bytecheck::CheckBytes<__C> for lindera_dictionary::viterbi::ArchivedWordId where u32: rkyv::traits::Archive, bool: rkyv::traits::Archive, lindera_dictionary::viterbi::LexType: rkyv::traits::Archive, <__C as rancor::Fallible>::Error: rancor::Trace, <u32 as rkyv::traits::Archive>::Archived: bytecheck::CheckBytes<__C>, <bool as rkyv::traits::Archive>::Archived: bytecheck::CheckBytes<__C>, <lindera_dictionary::viterbi::LexType as rkyv::traits::Archive>::Archived: bytecheck::CheckBytes<__C>
 pub unsafe fn lindera_dictionary::viterbi::ArchivedWordId::check_bytes(*const Self, &mut __C) -> core::result::Result<(), <__C as rancor::Fallible>::Error>
 pub struct lindera_dictionary::viterbi::Edge
-pub lindera_dictionary::viterbi::Edge::edge_type: lindera_dictionary::viterbi::EdgeType
-pub lindera_dictionary::viterbi::Edge::kanji_only: bool
-pub lindera_dictionary::viterbi::Edge::left_index: u16
-pub lindera_dictionary::viterbi::Edge::path_cost: i32
-pub lindera_dictionary::viterbi::Edge::start_index: u32
-pub lindera_dictionary::viterbi::Edge::stop_index: u32
-pub lindera_dictionary::viterbi::Edge::word_entry: lindera_dictionary::viterbi::WordEntry
 impl lindera_dictionary::viterbi::Edge
 pub fn lindera_dictionary::viterbi::Edge::num_chars(&self) -> usize
 impl core::clone::Clone for lindera_dictionary::viterbi::Edge
@@ -957,25 +932,17 @@
 impl core::default::Default for lindera_dictionary::viterbi::Lattice
 pub fn lindera_dictionary::viterbi::Lattice::default() -> lindera_dictionary::viterbi::Lattice
 pub struct lindera_dictionary::viterbi::PathEntry
-pub lindera_dictionary::viterbi::PathEntry::cost: i32
-pub lindera_dictionary::viterbi::PathEntry::edge_index: u16
-pub lindera_dictionary::viterbi::PathEntry::left_index: u16
-pub lindera_dictionary::viterbi::PathEntry::left_pos: u32
 impl core::clone::Clone for lindera_dictionary::viterbi::PathEntry
 pub fn lindera_dictionary::viterbi::PathEntry::clone(&self) -> lindera_dictionary::viterbi::PathEntry
 impl core::fmt::Debug for lindera_dictionary::viterbi::PathEntry
 pub fn lindera_dictionary::viterbi::PathEntry::fmt(&self, &mut core::fmt::Formatter<'_>) -> core::fmt::Result
 pub struct lindera_dictionary::viterbi::WordEntry
-pub lindera_dictionary::viterbi::WordEntry::left_id: u16
-pub lindera_dictionary::viterbi::WordEntry::right_id: u16
-pub lindera_dictionary::viterbi::WordEntry::word_cost: i16
-pub lindera_dictionary::viterbi::WordEntry::word_id: lindera_dictionary::viterbi::WordId
 impl lindera_dictionary::viterbi::WordEntry
-pub const lindera_dictionary::viterbi::WordEntry::SERIALIZED_LEN: usize
-pub fn lindera_dictionary::viterbi::WordEntry::deserialize(&[u8], bool) -> lindera_dictionary::viterbi::WordEntry
 pub fn lindera_dictionary::viterbi::WordEntry::left_id(&self) -> u32
+pub fn lindera_dictionary::viterbi::WordEntry::new(lindera_dictionary::viterbi::WordId, i16, u16, u16) -> Self
 pub fn lindera_dictionary::viterbi::WordEntry::right_id(&self) -> u32
-pub fn lindera_dictionary::viterbi::WordEntry::serialize<W: std::io::Write>(&self, &mut W) -> std::io::error::Result<()>
+pub fn lindera_dictionary::viterbi::WordEntry::word_cost(&self) -> i16
+pub fn lindera_dictionary::viterbi::WordEntry::word_id(&self) -> lindera_dictionary::viterbi::WordId
 impl core::clone::Clone for lindera_dictionary::viterbi::WordEntry
 pub fn lindera_dictionary::viterbi::WordEntry::clone(&self) -> lindera_dictionary::viterbi::WordEntry
 impl core::cmp::Eq for lindera_dictionary::viterbi::WordEntry
@@ -1002,10 +969,8 @@
 pub fn lindera_dictionary::viterbi::WordEntry::serialize(&self, &mut __S) -> core::result::Result<<Self as rkyv::traits::Archive>::Resolver, <__S as rancor::Fallible>::Error>
 pub struct lindera_dictionary::viterbi::WordEntryResolver where lindera_dictionary::viterbi::WordId: rkyv::traits::Archive, i16: rkyv::traits::Archive, u16: rkyv::traits::Archive
 pub struct lindera_dictionary::viterbi::WordId
-pub lindera_dictionary::viterbi::WordId::id: u32
-pub lindera_dictionary::viterbi::WordId::is_system: bool
-pub lindera_dictionary::viterbi::WordId::lex_type: lindera_dictionary::viterbi::LexType
 impl lindera_dictionary::viterbi::WordId
+pub fn lindera_dictionary::viterbi::WordId::id(&self) -> u32
 pub fn lindera_dictionary::viterbi::WordId::is_system(&self) -> bool
 pub fn lindera_dictionary::viterbi::WordId::is_unknown(&self) -> bool
 pub fn lindera_dictionary::viterbi::WordId::lex_type(&self) -> lindera_dictionary::viterbi::LexType
@@ -1036,5 +1001,6 @@
 pub fn lindera_dictionary::viterbi::WordId::serialize(&self, &mut __S) -> core::result::Result<<Self as rkyv::traits::Archive>::Resolver, <__S as rancor::Fallible>::Error>
 pub struct lindera_dictionary::viterbi::WordIdResolver where u32: rkyv::traits::Archive, bool: rkyv::traits::Archive, lindera_dictionary::viterbi::LexType: rkyv::traits::Archive
 pub fn lindera_dictionary::viterbi::is_kanji(char) -> bool
+pub macro lindera_dictionary::embedded_dictionary!
 pub fn lindera_dictionary::get_version() -> &'static str
 pub type lindera_dictionary::LinderaResult<T> = core::result::Result<T, lindera_dictionary::error::LinderaError>
